contents/docs/self-host/runbook/plugin-server/wtf.mdx

```diff
@@ -0,0 +1,289 @@
+3. Schedule periodic tasks (via `runEveryMinute`, `runEveryHour`, and `runEveryDay`)
+4. Trigger custom jobs to be run at any given point in the future (via the jobs API)
+5. Access APIs for key-value storage (using Postgres) and cache (using Redis)
+6. Create new events (via the `posthog` API inside plugin VMs)
+
+Beyond this, they can also use some Node APIs and third-party libraries. 
+
+To read about everything plugins can do, take a look at our [Developer Reference](https://posthog.com/docs/apps/build/reference).
+
+## Plugins vs. plugin configs
+
+There are two core models that plugins are built on top of - plugins (stored in the Postgres table `posthog_plugin` ) and plugin configs (stored in the Postgres table `posthog_pluginconfig`).
+
+Plugins represent plugin "definitions" - most importantly they contain the plugin's config schema, and the plugin code, as well as metadata such as name, description, and logo.
+
+From an installed plugin, teams in PostHog can set up their own instance of a plugin, called a "plugin config". 
+
+For instance, a plugin may depend on a configuration option called `eventToExport`. Then, when a team decides to enable that plugin, they will set `eventToExport` to the desired event name that suits their purpose, like `my_special_event`. 
+
+Plugin configs are what we use to setup plugin VMs. We load the plugin's code and inject the config's context.
+
+For example, all storage and cache usage will be indexed by config ID, and a given VM will only receive events and actions for the team associated with the plugin config.
+
+<HiddenSection headingType='h3' title='Advanced concepts' >
+
+Initially, for every plugin config, we would setup a plugin VM on **every thread on every server**.
+
+Since then, two things have changed:
+
+- **Stateless plugins:** Plugins marked as "stateless" use the same VM for all plugin configs. In other words, the same VM is used for all teams that have the plugin enabled. This is because they don't rely on any team-specific context. The GeoIP plugin, for example, takes an IP and enriches the event with some location data, without depending on any additional configuration. The same operation is done for all events across all teams, so there's no need to run a VM per config. [*Read more here*](https://github.com/PostHog/posthog/pull/8112).
+- **Capabilities:** A nice feature that came about as a result of plugin server "modes" is that we no longer need to run all plugins on all servers. For instance, if a plugin only has scheduled tasks, it will never be used on the ingestion server, so we don't need a VM for it on that server. [*Read more here*](https://github.com/PostHog/posthog/pull/9917).
+
+
+</HiddenSection>
+
+## Architecture
+
+### Event flow
+
+Here's a simplified high level overview of how an event flows through the PostHog ingestion pipeline:
+
+**Diagram 1: Event flow**
+
+![Event flow](../../../../images/plugin-server/event-flow.png)
```

Comment:
I strongly think we shouldn't include this as an image - it makes this doc unchangeable.

E.g. I had several suggestions on these graphs and I'm now unable to just fix it myself.

Reply:
added